### PR TITLE
ACM 10019 - enable managedHub Search

### PR DESCRIPTION
**Pull request by SherinV**
### Related Issue

https://issues.redhat.com/browse/ACM-10019

### Description of changes
- check if to proceed with Search based on ManagedHub filter

### CODE REVIEW

File: pkg/resolver/search.go [hunk 1]
```diff
@@ -77,6 +98,9 @@ func (s *SearchResult) Count() (int, error) {
 func (s *SearchResult) Items() ([]map[string]interface{}, error) {
 	s.wg.Add(1)
 	defer s.wg.Done()
+	if !s.isManagedHub() { // if current hub is not part of managedHub filter, stop search
+		return nil, nil
```
Comment (jlpadilla) [inline pkg/resolver/search.go:102]: There's a slight difference with returning nil vs. an empty []map[string]interface{}.  I think an empty list is slightly better and safer.

File: pkg/resolver/search.go [hunk 2]
```diff
@@ -65,7 +67,26 @@ func Search(ctx context.Context, input []*model.SearchInput) ([]*SearchResult, e
 
 }
 
+// Stop search if managedHub is in filters and current hub name is not in values.
+// Otherwise, proceed with the search.
+func (s *SearchResult) isManagedHub() bool {
```
Comment (jlpadilla) [inline pkg/resolver/search.go:72]: maybe a more specific name could be `matchesManagedHubFilter()`

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold

**Comment by SherinV** [comment]
Need to merge https://github.com/stolostron/search-v2-operator/pull/191 too.

**Pull request merged**